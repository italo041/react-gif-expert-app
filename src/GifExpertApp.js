import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";
import { Navbar } from "./components/Navbar";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <Navbar />
      <hr />
      <div className="container">
        <AddCategory setCategories={setCategories} />

        <ol>
          {categories.map((category) => (
            <GifGrid key={category} category={category} />
          ))}
        </ol>
      </div>
    </>
  );
};

export default GifExpertApp;
