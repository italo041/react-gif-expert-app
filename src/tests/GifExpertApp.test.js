import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import GifExpertApp from '../GifExpertApp';

describe('Prueba del componente <GifExpertApp/>', () => {
    

    test('Debe de mostrarse correctamente ', () => {

     const wrapper = shallow( <GifExpertApp /> );

     expect( wrapper ).toMatchSnapshot();

    })

    test('Debe de mostrar una lista de categorias', () => {
        
        const categories = ['One punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories= { categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    
})
