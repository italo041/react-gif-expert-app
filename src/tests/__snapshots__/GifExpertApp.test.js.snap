// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Prueba del componente <GifExpertApp/> Debe de mostrarse correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
